File: algorithms/no_repeats/no_repeats.test.js
/*
 * Created on Fri Apr 30 2021
 *
 * Copyright (c) 2021 - Joshua Sizer
 *
 * This code is licensed under MIT license (see
 * LICENSE for details)
 */

const permAlone = require("./no_repeats")

test('permAlone("aab")', () => {
    expect(permAlone("aab")).toEqual(2)
})

test('permAlone("aabb")', () => {
    expect(permAlone("aabb")).toEqual(8)
})

test('permAlone("abcdefa")', () => {
    expect(permAlone("abcdefa")).toEqual(3600)
})

test('permAlone("abfdefa")', () => {
    expect(permAlone("abfdefa")).toEqual(2640)
})

test('permAlone("zzzzzzzz")', () => {
    expect(permAlone("zzzzzzzz")).toEqual(0)
})

test('permAlone("a")', () => {
    expect(permAlone("a")).toEqual(1)
})

test('permAlone("aaab")', () => {
    expect(permAlone("aaab")).toEqual(0)
})

test('permAlone("aaabb")', () => {
    expect(permAlone("aaabb")).toEqual(12)
})

File: algorithms/no_repeats/no_repeats_test.py
"""
Created on Fri Apr 30 2021

Copyright (c) 2021 - Joshua Sizer

This code is licensed under MIT license (see
LICENSE for details)
"""

from no_repeats import perm_alone

class TestPermAlone:

    def test_perm_alone1(self):
        assert perm_alone("aab") == 2

    def test_perm_alone2(self):
        assert perm_alone("aabb") == 8

    def test_perm_alone3(self):
        assert perm_alone("abcdefa") == 3600

    def test_perm_alone4(self):
        assert perm_alone("abfdefa") == 2640

    def test_perm_alone5(self):
        assert perm_alone("zzzzzzzz") == 0

    def test_perm_alone6(self):
        assert perm_alone("a") == 1

    def test_perm_alone6(self):
        assert perm_alone("aaab") == 0

    def test_perm_alone6(self):
        assert perm_alone("aaabb") == 12